php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.VARIABLE_WITHOUT_OBJECTS).is(
+      f.variableWithoutObjects(
+        b.zeroOrMore(b.token(PHPLexicalGrammar.VARIABLE_VARIABLE_DOLLAR)),
+        COMPOUND_VARIABLE(),
+        b.zeroOrMore(b.firstOf(
+          DIMENSIONAL_OFFSET(),
+          ALTERNATIVE_DIMENSIONAL_OFFSET()))));
+  }
+
+  public ArrayAccessTree ALTERNATIVE_DIMENSIONAL_OFFSET() {
+    return b.<ArrayAccessTree>nonterminal().is(
+      f.alternativeDimensionalOffset(
+        b.token(LCURLYBRACE),
+        b.optional(EXPRESSION()),
+        b.token(RCURLYBRACE)));
+  }
+
+  public ArrayAccessTree DIMENSIONAL_OFFSET() {
+    return b.<ArrayAccessTree>nonterminal(PHPLexicalGrammar.DIMENSIONAL_OFFSET).is(
+      f.dimensionalOffset(
+        b.token(LBRACKET),
+        b.optional(EXPRESSION()),
+        b.token(RBRACKET)));
+  }
+
+  public ExpressionTree PRIMARY_EXPRESSION() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.PRIMARY_EXPRESSION).is(
+      b.firstOf(
+        f.newStaticIdentifier(b.token(STATIC)),
+        NAMESPACE_NAME(),
+        VARIABLE_WITHOUT_OBJECTS(),
+        IDENTIFIER(),
+        PARENTHESIZED_EXPRESSION()));
+  }
+
+  public AssignmentByReferenceTree ASSIGNMENT_BY_REFERENCE() {
+    return b.<AssignmentByReferenceTree>nonterminal(PHPLexicalGrammar.ASSIGNMENT_BY_REFERENCE).is(
+      f.assignmentByReference(
+        MEMBER_EXPRESSION(),
+        b.token(PHPPunctuator.EQU), b.token(AMPERSAND),
+        b.firstOf(NEW_EXPRESSION(), MEMBER_EXPRESSION())));
+  }
+
+  public AssignmentExpressionTree ASSIGNMENT_EXPRESSION() {
+    return b.<AssignmentExpressionTree>nonterminal(PHPLexicalGrammar.ASSIGNMENT_EXPRESSION).is(
+      f.assignmentExpression(MEMBER_EXPRESSION(), ASSIGNMENT_OPERATOR(), EXPRESSION()));
+  }
+
+  public InternalSyntaxToken ASSIGNMENT_OPERATOR() {
+    return b.<InternalSyntaxToken>nonterminal(PHPLexicalGrammar.ASSIGNMENT_OPERATOR).is(
+      b.firstOf(
+        b.token(PHPPunctuator.EQU),
+
+        b.token(PHPPunctuator.STAR_EQU),
+        b.token(PHPPunctuator.DIVEQUAL),
+        b.token(PHPPunctuator.MOD_EQU),
+        b.token(PHPPunctuator.PLUS_EQU),
+        b.token(PHPPunctuator.MINUS_EQU),
+        b.token(PHPPunctuator.SL_EQU),
+        b.token(PHPPunctuator.SR_EQU),
+        b.token(PHPPunctuator.CONCATEQUAL),
+
+        b.token(PHPPunctuator.ANDEQUAL),
+        b.token(PHPPunctuator.XOR_EQU),
+        b.token(PHPPunctuator.OR_EQU)));
+  }
+
+  public ReferenceVariableTree REFERENCE_VARIABLE() {
+    return b.<ReferenceVariableTree>nonterminal(Kind.REFERENCE_VARIABLE).is(
+      f.referenceVariable(
+        b.token(AMPERSAND),
+        MEMBER_EXPRESSION()));
+  }
+
+  public SpreadArgumentTree SPREAD_ARGUMENT() {
+    return b.<SpreadArgumentTree>nonterminal(Kind.SPREAD_ARGUMENT).is(
+      f.spreadArgument(
+        b.token(ELIPSIS),
+        EXPRESSION()));
+  }
+
+  public FunctionCallTree FUNCTION_CALL_ARGUMENT_LIST() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)